examples/weather-demo/README.md

```diff
@@ -0,0 +1,77 @@
+<!-- Copyright © SixtyFPS GmbH <info@slint.dev> ; SPDX-License-Identifier: MIT -->
+
+# Rusty Weather
+
+Rusty Weather is a weather application made by [Felgo](https://felgo.com/).
+
+The application retrieves weather data from the [OpenWeather](https://openweathermap.org/) API to provide:
+* Real-time weather data,
+* 8-day forecasts data,
+* Temperatures at particular times of the day,
+* Daily rain amount and rain probability,
+* UV index level,
+* support for various locations around the globe.
+
+ The project demonstrates how to write a cross-platform Rust GUI application using the [Slint](https://slint.dev/) toolkit.
+ It includes subjects like:
+ * responsive layouts and adaptions for different screen sizes and orientations,
+ * providing fully customized look and feel with custom widgets,
+ * integrating Slint code with the Rust backend code,
+ * using `async` features with a Slint-based application and combining it with the Tokio runtime.
+
+<br />
+<p>
+  <img src="docs/img/desktop-preview.png" height="415px" />
+  <img src="docs/img/android-preview.png" height="415px" hspace="5" />
+</p>
+
+## Weather data
+To enable real weather data from the [OpenWeather](https://openweathermap.org/) API, you must provide the `RUSTY_WEATHER_API_KEY` environment variable with your API key at build time. The [OpenCall API](https://openweathermap.org/price#onecall) subscription is required.
```

Comment:
This thought is beyond the scope of this PR: We have similar needs in the energy monitor demo, where we use weatherapi.com and we've got an API key in the CI set, so that the demos built have weather data support. It might make sense to unify these, i.e. either to switch the energy monitor over to use openweathermap.org or the other way around. The key from weatherapi.com is free.

Reply:
That sounds like a good idea :slightly_smiling_face: 

Just so you know, the OpenWeather API is also mostly free. Only the OneCall API we used requires a paid subscription, and it is only paid for calls above 1000 per day.

The other APIs available for free keys can provide similar functionalities. We only switched to the OneCall version because we were facing issues. However, as we later discovered, those were issues in the crate we used rather than in the API itself.